src/xunit.analyzers/AssertIsTypeShouldNotBeUsedForAbstractTypeFixer.cs

```diff
@@ -0,0 +1,51 @@
+﻿using System.Collections.Immutable;
+using System.Composition;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CodeActions;
+using Microsoft.CodeAnalysis.CodeFixes;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Editing;
+
+namespace Xunit.Analyzers
+{
+    [ExportCodeFixProvider(LanguageNames.CSharp), Shared]
+    public class AssertIsTypeShouldNotBeUsedForAbstractTypeFixer : CodeFixProvider
+    {
+        private const string Title = "Use Assert." + IsAssignableFrom;
+
+        private const string IsAssignableFrom = "IsAssignableFrom";
+
+        private const string IsType = "IsType";
+
+        public sealed override ImmutableArray<string> FixableDiagnosticIds { get; } = ImmutableArray.Create(Descriptors.X2017_AssertIsTypeShouldNotBeUsedForAbstractType.Id);
+
+        public sealed override FixAllProvider GetFixAllProvider() => WellKnownFixAllProviders.BatchFixer;
+
+        public sealed override async Task RegisterCodeFixesAsync(CodeFixContext context)
+        {
+            var root = await context.Document.GetSyntaxRootAsync(context.CancellationToken).ConfigureAwait(false);
+            var invocation = root.FindNode(context.Span).FirstAncestorOrSelf<InvocationExpressionSyntax>();
+            var methodName = invocation.GetGenericName()?.Identifier.Text;
+            if (methodName == IsType)

```

Comment:
I have changed my mind on that last sentence-- it's easy for the user to just press Ctrl+X if he needs to delete the assert. I don't feel it's necessary to write a code fix provider for that in this PR.